src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.tsx

```diff
@@ -186,7 +186,12 @@ const ElementsSection = (props: {
         <DataElementsTable
           allowedTypes={allowedTypes}
           dataElements={filterDataElements(patient?.dataElements)}
-          onView={(dataElement) => setSelectedDataElement(dataElement)}
+          onView={(dataElement) => {
+            const modelClass = getDataElementClass(dataElement);
+            const updatedDataElement = new modelClass(dataElement);
```

Comment:
can you check once if the `id` remains same. I think should be fine just making sure it doesn't change.